### ISSUE TITLE
x86 parser for repz scasl

### ISSUE THREAD

**Issue by LRGH** [bdcht/amoco#86]
```
from amoco.arch.x86 import cpu_x86 
from amoco.ui import render

cpu_x86.instruction_x86.set_formatter(cpu_x86.IA32_Binutils_ATT)
render.conf.UI.formatter = 'Null'
        
s = b'\xf3\xaf'
i = cpu_x86.disassemble(s)
assert str(i) == 'repz scasl     '
assert i.length == 2
```
The bug is that `i.length == 3` in recent amoco.

**Comment by LRGH** [comment]
I think that the following patch corrects the bug.
```
--- a/amoco/arch/core.py
+++ b/amoco/arch/core.py
@@ -617,12 +617,12 @@ class ispec(object):
         # create & update instruction object:
         if i is None:
             i = iclass(bs)
-        else:
-            i.bytes += bs
         # check any precondition on i:
         if self.precond:
             if not self.precond(i):
                 raise DecodeError
+        if i.bytes != bs:
+            i.bytes += bs
         # ok, lets call the spec hook...
         i.spec = self
         # set instruction attributes from directives, and then
```

**Comment by bdcht** [comment]
my mistake...the precond verifier needed to be placed in the try...except clause to get rid of the added bytes in case of InstructionError. Fixed by commit a0f4b6a.
